src/vl-map-layer-switcher.js

```diff
@@ -103,7 +115,21 @@ export class VlMapLayerSwitcher extends vlElement(HTMLElement) {
     }
   }
 
-  _getLayer(input) {
-    return this._map.getOverlayLayers().find((layer) => layer.get('title') == input.dataset.vlLayer);
+  _computeInputsDisabledAttribute() {
+    const resolution = this._map.getView().getResolution();
+    this._layerInputs.forEach((input) => this._computeInputDisabledAttribute(input, resolution));
+  }
+
+  _computeInputDisabledAttribute(input, resolution) {
+    const layer = this._getLayer(input);
+    if (layer) {
+      const maxResolution = parseFloat(layer.getMaxResolution());
+      const minResolution = parseFloat(layer.getMinResolution());
+      if (resolution >= maxResolution || resolution < minResolution) {
+        input.setAttribute('disabled', '');
+      } else {
+        input.removeAttribute('disabled');
+      }
+    }
```

Comment:
ge werkt hier altijd tegen de "openlayers" layers en niet tegen de "vl-ui" layers?
is het niet "logischer" om tegen de vl-ui te werken? dan zou je evt ook kunnen nu een methode daarop maken met "isShownAtResolution(resolution)" of zo ...

Reply:
encapsulatie for the win als het ware :-)

Reply:
De OL map is een publieke getter op <vl-map/> en is daardoor eigenlijk onderdeel van de API. Een gebruiker kan daardoor ook rechtstreeks wijzigingen uitvoeren via OL. Via de vl-ui API kunnen we natuurlijk wel zoveel als mogelijk relevante functionaliteit aanbieden, dus hier kan ik inderdaad logica verhuizen en zal dat uitvoeren.

Reply:
ja, ideaal gezien "wrapped" ge volledig alles, en encapsuleert ge volledig alles van OL ...
maar "zoveel als praktisch mogelijk" lijkt me ook al wel goed :-)